lib/foreman_maintain/cli/upgrade_command.rb

```diff
@@ -52,12 +52,23 @@ def print_versions(target_versions)
         target_versions.sort.each { |version| puts version }
       end
 
+      def allow_self_upgrade?
+        if disable_self_upgrade?
+          return !disable_self_upgrade?
```

Comment:
```suggestion
          return false
```

Easier to read ;-) (also, I think the `return` is *technically* not necessary, and you don't `return` in the other branches)